src/NServiceBus.Extensions.Diagnostics/OutgoingPhysicalMessageDiagnostics.cs

```diff
@@ -13,41 +13,32 @@ public class OutgoingPhysicalMessageDiagnostics : Behavior<IOutgoingPhysicalMess
         private const string EventName = ActivityNames.OutgoingPhysicalMessage + ".Sent";
 
         public OutgoingPhysicalMessageDiagnostics(IActivityEnricher activityEnricher)
+            : this(new DiagnosticListener(ActivityNames.OutgoingPhysicalMessage), activityEnricher)
         {
-            _diagnosticListener = new DiagnosticListener(ActivityNames.OutgoingPhysicalMessage);
+        }
+
+        public OutgoingPhysicalMessageDiagnostics(DiagnosticListener diagnosticListener,
+            IActivityEnricher activityEnricher)
+        {
+            _diagnosticListener = diagnosticListener;
             _activityEnricher = activityEnricher;
         }
 
         public override async Task Invoke(IOutgoingPhysicalMessageContext context, Func<Task> next)
         {
-            using (var activity = StartActivity(context))
+            var activity = Activity.Current;
```

Comment:
I think this is OK but one thing I found in other places is you can't necessarily assume Activity.Current is there.